service/create/service.go

```diff
@@ -912,6 +914,28 @@ func (s *Service) processCluster(cluster awstpr.CustomObject) error {
 	}
 
 	// Create an Auto Scaling Group for the workers.
+	asg := awsresources.AutoScalingGroup{
+		Client:    clients.AutoScaling,
+		Name:      key.AutoScalingGroupName(cluster, prefixWorker),
+		ClusterID: key.ClusterID(cluster),
+	}
+	legacyASG, err := asg.FindLegacy()
+	if err != nil {
+		return microerror.Maskf(executionFailedError, fmt.Sprintf("search for legacy worker ASG failed: '%#v'", err))
+	}
+
+	var instances []*autoscaling.Instance
```

Comment:
Can we somehow don't reuse `instances` variable? Otherwise this code is not idempotent. When CF fails after deleting legacy ASG we have orphaned instances. If the code was selecting detached instances when the legacy ASG is gone after executing CF the problem would be solved.

Reply:
mm not sure, there could be other detached instances not related to any ASG and without any reference to the cluster, how could we identify the ones to attach?

Reply:
I just thought we could make sure there is a tag on each instance before detaching it. That way we could get rid of the backoff in detaching. WDYT?

Reply:
Totally, i was thinking the same, adding tags to instances would help a lot. I'm thinking of wrapping all the code on a separate component, there's a lot of things going on currently in the migration. This way we could reuse/adapt it in the future, will ping you when ready.

Reply:
Per instance tags would be very useful. We have node IDs in the cluster TPO but these aren't applied to the instances because the tags are from the autoscaling group.

These would be useful for the API so a user could terminate a specific node.